NXP Kinetis: Only perform early watchdog initialisation once

henrikbrixandersen opened the pull request:
Currently, early watchdog initialisation (by calling `z_arm_watchdog_init`) is done by both the boot loader and the application. Early watchdog initialisation should be just that (early) and must happen a SoC specific number of clock cycles after reset.

This series of patches unifies the `CONFIG_WDOG_INIT` symbol across the NXP Kinetis series and disables it for applications to be chainloaded by MCUboot.

This has been tested to work with the `twr_ke18f` and `frdm_k64f` boards.

henrikbrixandersen committed:
soc: nxp: kinetis: kl2x: move watchdog init to z_arm_watchdog_init

Move the NXP Kinetis KL2x watchdog initialization code to
z_arm_watchdog_init() and make it optional based on
CONFIG_WDOG_INIT.

This brings the KL2x in line with the other NXP Kinetis series SoC
implementations.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

henrikbrixandersen committed:
soc: arm: nxp: kinetis: unify CONFIG_WDOG_INIT Kconfig option

Move CONFIG_WDOG_INIT for the NXP Kinetis series to the top-level
Kinetis Kconfig file and enable it where needed.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

henrikbrixandersen commented on soc/arm/nxp_kinetis/k2x/Kconfig.soc:
```diff
@@ -71,10 +71,6 @@ config K22_FLASH_CLOCK_DIVIDER
-	  This processor enables the watchdog timer with a short timeout
-	  upon reset. Therefore, this requires that the watchdog be configured
-	  during reset handling.
+	default y
```
It is currently not enabled for (at least) `CONFIG_SOC_MKW40Z4` and `CONFIG_SOC_MKW41Z4`. I can set the default to `y` in the top-level and override it for those instead, if preferred?

MaureenHelm replied:
It looks like kw4xz needs the same change you made in `soc: nxp: kinetis: kl2x: move watchdog init to z_arm_watchdog_init`. Would you mind doing this too and make all the defaults the same?

henrikbrixandersen replied:
@MaureenHelm Done. I have only compile-tested it.

MaureenHelm replied:
Thanks. I tested it on frdm_kw41z

henrikbrixandersen committed:
soc: nxp: kinetis: kw4xz: move watchdog init to z_arm_watchdog_init

Move the NXP Kinetis KW4xZ watchdog initialization code to
z_arm_watchdog_init() and make it optional based on CONFIG_WDOG_INIT.

This brings the KW4xZ in line with the other NXP Kinetis series SoC
implementations.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

henrikbrixandersen committed:
soc: arm: nxp: kinetis: disable watchdog init if bootloader is enabled

Disable the early watchdog initialization for the NXP Kinetis series
if the application is to be chain-loaded by mcuboot.

The early watchdog initialization must only take place once and needs
to happen within a SoC specific number of CPU clock cycles after
reset.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>
